frontend/src/screens/GameScreen/gameScreen.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useEffect, useState } from "react";
+import { View, Text, StyleSheet } from "react-native";
+import { BLUE, GREY } from "../../constants/colors";
+import FirebaseInteractor from "../../firebase/firebaseInteractor";
+import { UID } from "../../models/types";
+import { durstenfeldShuffle } from "../../utils/utils";
+import { DraxView, DraxProvider } from "react-native-drax";
+import DroppableRow from "../../components/DroppableRow";
+import { useNavigation } from "@react-navigation/core";
+import { LoadingScreen } from "../../components/LoadingScreen";
+import PrimaryButton from "./PrimaryButton";
+import SecondaryButton from "./SecondaryButton";
+
+const fi = new FirebaseInteractor();
+
+interface MaybeWordPair {
+    english?: string;
+    turkish: string;
+    correctEnglishWord: string;
+}
+
+export interface GameScreenProps {
+  route: any;
+}
+
+export interface SpecificGameScreenProps {
+    route: any;
+    shuffleFunction: (words: any[]) => any[];
+    topScreenRenderFunction: (numCorrect: number, numTotal: number) => JSX.Element;
+    isPairing: boolean;
+}
+
+export default function GameScreen(props: SpecificGameScreenProps) {
+    const [englishWords, setEnglishWords] = useState<string[]>()
+    const [turkishWords, setTurkishWords] = useState<MaybeWordPair[]>()
+    const [submitted, setSubmitted] = useState(false);
+    const navigation = useNavigation()
+    const { sessionId } = props.route.params;
+    const [currentRoundId, setCurrentRoundId] = useState<UID>("");
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+        console.log("sessionId = " + sessionId)
+        if (currentRoundId == "") {
+            fi.startRound(sessionId).then(result => setCurrentRoundId(result));
+        } else {
+            fi.getUser().then(user => {
+                fi.getRoundPairs(currentRoundId).then(words => {
+                    setEnglishWords(durstenfeldShuffle(props.shuffleFunction(words)));
+                    setTurkishWords(durstenfeldShuffle(words.map((word, i) => ({ turkish: word.turkish, correctEnglishWord: word.english, english: undefined }))));
+                    setSubmitted(false)
+                    setIsLoading(false)
+                }).catch(console.error);
+            }).catch(console.error);
+        }
+    }, [currentRoundId]);
+
+    if (englishWords === undefined) {
+        return <LoadingScreen />
+    }
+
+    const restartRound = () => {
+        fi.endRound(currentRoundId);
+        fi.startRound(sessionId).then(result => setCurrentRoundId(result));
+    }
+
+    const renderEnglishOptions = () => {
```

Comment:
This is nitpicky, but could we move this method down to be with the other render methods?